syft/execution/role.py

```diff
@@ -116,6 +122,20 @@ def _instantiate_inputs(self, args):
         )
         PlaceHolder.instantiate_placeholders(input_placeholders, args)
 
+        # Last extra argument is a state?
+        if (
+            self.state is not None
```

Comment:
nit: self.state should never be None (in the constructor, it is set as a new Role if none is provided)